enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/JWTAuthenticator.java

```diff
@@ -164,6 +188,63 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
+            throws APISecurityException {
+        log.debug("Inside generateAndRetrieveJWTToken");
+
+        String endUserToken = null;
+        boolean valid = false;
+        String jwtTokenCacheKey = jwtInfoDto.getApicontext().concat(":").concat(jwtInfoDto.getVersion()).concat(":")
+                .concat(tokenSignature);
+        JWTConfigurationDto jwtConfigurationDto = ConfigHolder.getInstance().getConfig().getJwtConfigurationDto();
+        // Get the jwt generator class (Default jwt generator class)
+        //Todo: load the class from the configuration
+        jwtGenerator = JWTUtil.getApiMgtGatewayJWTGenerator();
+        if (jwtGenerator != null) {
+            jwtGenerator.setJWTConfigurationDto(jwtConfigurationDto);
+            if (isGatewayTokenCacheEnabled) {
+                try {
+                    Object token = CacheProvider.getGatewayJWTTokenCache().get(jwtTokenCacheKey);
+                    if (token != null) {
+                        endUserToken = (String) token;
+                        String[] splitToken = ((String) token).split("\\.");
+                        org.json.JSONObject payload = new org.json.JSONObject(new String(Base64.getUrlDecoder().
+                                decode(splitToken[1])));
+                        long exp = payload.getLong("exp");
+                        long timestampSkew = getTimeStampSkewInSeconds() * 1000;
+                        valid = (exp - System.currentTimeMillis() > timestampSkew);
+                    }
+                } catch (Exception e) {
+                    log.error("Error while getting token from the cache", e);
+                }
+
+                if (org.apache.commons.lang.StringUtils.isEmpty(endUserToken) || !valid) {
+                    try {
+                        endUserToken = jwtGenerator.generateToken(jwtInfoDto);
+                        CacheProvider.getGatewayJWTTokenCache().put(jwtTokenCacheKey, endUserToken);
+                    } catch (JWTGeneratorException e) {
+                        log.error("Error while Generating Backend JWT", e);
+                        throw new APISecurityException(APISecurityConstants.API_AUTH_GENERAL_ERROR,
+                                APISecurityConstants.API_AUTH_GENERAL_ERROR_MESSAGE, e);
+                    }
+                }
+            } else {
+                try {
+                    endUserToken = jwtGenerator.generateToken(jwtInfoDto);
+                } catch (JWTGeneratorException e) {
+                    log.error("Error while Generating Backend JWT", e);
+                    throw new APISecurityException(APISecurityConstants.API_AUTH_GENERAL_ERROR,
+                            APISecurityConstants.API_AUTH_GENERAL_ERROR_MESSAGE, e);
+                }
+            }
+        } else {
+            log.debug("Error in loading class");
+            System.out.print("Error in loading class");
```

Comment:
Remove System.out statements. 
Wonder how this is not checked from checkstyle plugin